src/diffusers/models/transformer_2d.py

```diff
@@ -228,11 +230,15 @@ def forward(
             encoder_hidden_states ( `torch.LongTensor` of shape `(batch size, encoder_hidden_states dim)`, *optional*):
                 Conditional embeddings for cross attention layer. If not given, cross-attention defaults to
                 self-attention.
-            timestep ( `torch.long`, *optional*):
+            timestep ( `torch.LongTensor`, *optional*):
                 Optional timestep to be applied as an embedding in AdaLayerNorm's. Used to indicate denoising step.
             class_labels ( `torch.LongTensor` of shape `(batch size, num classes)`, *optional*):
                 Optional class labels to be applied as an embedding in AdaLayerZeroNorm. Used to indicate class labels
                 conditioning.
+            attention_mask ( `torch.Tensor` of shape (batch size, num latent pixels), *optional* ).
+                Bias to add to attention scores.
```

Comment:
This is not the bias, it's an `attention_mask` which says which positions should be masked out no?

Reply:
can you elaborate on how you came to that conclusion?

to my eye: the only place where masks are promoted into biases, is at the very beginning of `UNet2DConditionModel#forward`. we are past that. any "1s and 0s" style mask, has already been turned into a "0s and -10000s" bias by the time we get here.

from here, `transformer_2d` passes this bias unchanged to `BasicTransformerBlock#forward`, which passes it unchanged to `Attention#forward`, which passes it unchanged to `AttnProcessor#forward`, which reshapes it, then adds it to the attention scores.

on the basis that it is being used for _offsetting attention scores via addition_: I call it a bias. if anything I feel it'd be more correct to rename all the parameters along the way to `attention_bias`.  
I figured that wouldn't fly, so I settled for at least documenting it.

Reply:
Ah you're right, the Transformer2D block is directly called in UNet2D. Ok in this case, we need to allow `attention_mask` to be both:
- of len(shape) = 2 and boolean
- and of len(shape) = 3 and float which is after this happens: https://github.com/huggingface/diffusers/blob/efc48da23bd342ca79a79def02649b1975393ea0/src/diffusers/models/unet_2d_condition.py#L655